syntax = "proto3";

package user;

service User {
  rpc CheckUserToken (UserRequestToken) returns (UserResponse) {}
  rpc CheckUserID (UserRequestID) returns (UserResponse) {}
}

message UserRequestToken {
  string token = 1;
}

message UserRequestID {
  string id = 1;
}

message UserResponse {
  uint64 user_id = 1;
  string username = 2;
  string email = 3;
  string name = 4;
  string role = 5;
}
